apps/app/src/app/shared/filters/task-status-filter/task-status-filter.component.spec.ts

```diff
@@ -0,0 +1,78 @@
+import { HttpClientModule } from '@angular/common/http';
+import { EventEmitter } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { TaskStatus } from '@armonik.admin.gui/armonik-typing';
+import { ClarityModule } from '@clr/angular';
+import { TranslateModule } from '@ngx-translate/core';
+import { TaskStatusFilterComponent } from './task-status-filter.component';
+
+describe('TaskStatusFilterComponent', () => {
+  let component: TaskStatusFilterComponent;
+  let fixture: ComponentFixture<TaskStatusFilterComponent>;
+
+  beforeEach(async () => {
+    TestBed.configureTestingModule({
+      declarations: [TaskStatusFilterComponent],
+      imports: [TranslateModule.forRoot(), ClarityModule],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TaskStatusFilterComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should have a property "name"', () => {
+    expect(component.name).toBeDefined();
+  });
+
+  it('should return the name with "property"', () => {
+    component.name = 'test';
+    expect(component.property).toBe(component.name);
+  });
+
+  it('should emit an event when the value changes', () => {
+    const changes = { emit: jasmine.createSpy('changes') };
+    component.changes = changes as unknown as EventEmitter<boolean>;
+    component.onChange({ target: { value: 'test' } });
+    expect(changes.emit).toHaveBeenCalled();
+  });
+
+  it('should change the selected value', () => {
+    component.onChange({ target: { value: TaskStatus.CANCELLED } });
+    expect(component.selectedValue).toBe(TaskStatus.CANCELLED);
+  });
+
+  it('should change the value', () => {
```

Comment:
Je pense qu'ici tu peux juste tester que value est bien égale à selected value, comme tu as fait entre property et name

Reply:
done

Reply:
Hmm je ne pense pas qu'il faille que tu changes le test "should change the selected value". Tu peux laisser la ligne 47 à `expect(component.selectedValue).toBe(TaskStatus.CANCELLED);`

Ce que je voulais dire surtout c'est que tu n'es pas obligé de tester le fait que value change avec ton test "should change the value" mais plutôt tester que ton getter value te retourne bien component.selectedValue. Et donc à fortiori si selectedValue change, ton getter value change. Mais ici on veut tester les fonctions de manière unitaires et pas forcément avec du contexte.